src/gltfPreviewDocumentContentProvider.ts

```diff
@@ -15,8 +15,60 @@ export class GltfPreviewDocumentContentProvider implements TextDocumentContentPr
         return 'file://' + this._context.asAbsolutePath(file);
     }
 
+    // credit: http://stackoverflow.com/a/30714706
+    private isAbsolutePath(pathToTest : string) : boolean {
+        return path.normalize(pathToTest + '/') === path.normalize(path.resolve(pathToTest) + '/');
+    }
+
+    /**
+    * @function fixPaths
+    * Does an in-place update of the provided gltfObject to ensure that all 'uri' properties are absolute paths.
+    * @param  {object} gltfObject The glTF content in object form
+    * @param  {string} gltfRootPath The root path that the glTF object is located in (which is where any relative binary files might be referenced from)
+    */
+    private fixPaths(gltfObject: object, gltfRootPath: string) {
+        if (!gltfRootPath.endsWith("\\"))
+        {
+            gltfRootPath = gltfRootPath + "\\";
+        }
+
+        for (var property in gltfObject) {
+            if (gltfObject.hasOwnProperty(property)) {
+                if (typeof gltfObject[property] == "object") {
+                    this.fixPaths(gltfObject[property], gltfRootPath);
+                } else if (property === "uri") {
+                    if (!gltfObject[property].startsWith('data:') && !this.isAbsolutePath(gltfObject[property]))
+                    {
+                        gltfObject[property] = gltfRootPath + gltfObject[property];
+                    }
+                }
+            }
+        }
+    }
+
     public provideTextDocumentContent(uri: Uri): string {
-        const glTF = vscode.window.activeTextEditor.document.getText();
+        const gltfRootPath = path.dirname(vscode.window.activeTextEditor.document.fileName);
+        const gltfFileName = path.basename(vscode.window.activeTextEditor.document.fileName);
+
+        // The "uri" property within a glTF file provides the reference to the binary content needed
+        // for the glTF to be processed.  The content of that property is likely to fall into one of
+        // four categories:
+        // 1. Just the binary filename
+        // 2. A partial path to the file.
+        // 3. A full path (or full Uri reference)
+        // 4. The full binary content base-64 encoded with a content-type header.
+        //
+        // If we don't make any changes to the glTF content, #3 and #4 work just fine with 3D renderers.
+        // Unfortunately, #1 and #2 are considered to be paths relative to the "pages" subfolder of this
+        // extension, and so those references fail to load.  So, we'll look for any "uri" property that
+        // doesn't start with "data" (which would make it #4) and test the path to see if it's absolute
+        // or not.  If it's not absolute, we'll prepend the root folder that the glTF file content
```

Comment:
Right.  The same rendered HTML won't work directly in Chrome due to the same origin policy unless you launch chrome with `--allow-file-access-from-files` (as suggested by by [this](https://threejs.org/docs/#manual/introduction/How-to-run-thing-locally)).  I was surprised it was working as-is within VS Code, but I wasn't going to argue.

As discussed in the other comment thread, I'm looking into the protocol solution for accessing the editor file content.